src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,76 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String PARSED_STRING_SEPARATOR = ";";
+    private static final String FORMED_STRING_SEPARATOR = ",";
+    private static final String BUY_CATEGORY_NAME = "buy";
+    private static final String SUPPLY_CATEGORY_NAME = "supply";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] data = readDataFromFile(fromFileName);
+        String report = formReport(data);
+        createNewDataFile(toFileName, report);
+    }
+
+    public String[] readDataFromFile(String fromFileName) {
+        try (FileReader fileReader = new FileReader(fromFileName)) {
+            StringBuilder parsedCsv = new StringBuilder();
+            BufferedReader reader = new BufferedReader(fileReader);
+
+            String value;
+            if ((value = reader.readLine()) == null) {
+                return null;
+            }
+            do {
+                parsedCsv.append(value).append(PARSED_STRING_SEPARATOR);
+            } while ((value = reader.readLine()) != null);
+
+            return parsedCsv.toString().split(PARSED_STRING_SEPARATOR);
+        } catch (IOException e) {
+            throw new RuntimeException("Try to read file was unsuccessful", e);
```

Comment:
make this message more informative. It will be great to add an input parameter (fromFileName) to the message.